src/index.js

```diff
@@ -40,11 +38,13 @@ export default (oldState, newState) => {
         } else if (!!value && isFunction(value.mergeDeep)) {
             result[key] = fromJS(oldValue).mergeDeep(value);
         } else if (isObject(value) && !isArray(value)) {
-            result[key] = merge({}, oldValue, value);
+            result[key] = immutableMerger(oldValue, value);
         } else {
             result[key] = value;
         }
     }
 
     return result;
-};
+}
+
+export default (oldState, newState) => immutableMerger(oldState, newState);
```

Comment:
Just move the `export default` up to the function definition and we can save this additional function layer :smile: